File: Board/src/main/java/com/example/board/domain/auth/domain/RefreshToken.java
package com.example.board.domain.auth.domain;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;
import org.springframework.data.redis.core.index.Indexed;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@RedisHash
public class RefreshToken {

    @Id
    private String userId;

    @Indexed
    private String refreshToken;

    @TimeToLive //토큰 만료시간을 seconds 단위로 설정핫기  위한 어노테이션
    private Long timeToLive;

    @Builder
    public RefreshToken(String accountId, String refreshToken) {
        this.userId = accountId;
        this.refreshToken = refreshToken;
        this.timeToLive = 3600L *2;
    }

    public void updateRefreshToken(String refreshToken, Long timeToLive) {
        this.refreshToken = refreshToken;
        this.timeToLive = timeToLive;
    }
}

File: Kopring/Dise_V2/src/main/kotlin/com/example/dise_v2/domain/user/domain/RefreshToken.kt
package com.example.dise_v2.domain.user.domain

import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.index.Indexed
import javax.persistence.Id
import javax.validation.constraints.NotBlank

@RedisHash(value = "RefreshToken", timeToLive = 60 * 60 * 2)
class RefreshToken(

    @Id
    val accountId: String,

    @Indexed
    @field:NotBlank
    val token: String
)